### PR TITLE
Jcessna/class refs

### PR DESCRIPTION
**Pull request by jcessna**
cleaned up all of the class ref/string business
also cleaned up (for the most part) serialization of tags.  we might need to look at how tags are parsed from the command line, if that api is used and consistent
cleaned up setup.py dependencies to provide clean installs without warnings
modified all existing tests to pass under new updates
-passing a 'pkg.mod.classname' str to apply is still supported, but deprecated.  you cannot pass an unqualified 'classname' string

### CODE REVIEW

File: disdat/apply.py [hunk 1]
```diff
@@ -51,7 +51,7 @@ def apply(output_bundle, pipe_params, pipe_cls, input_tags, output_tags, force,
     Args:
         output_bundle: The new bundle to be created
         pipe_params (str):   Luigi Task parameters string
```
Comment (kyocum) [inline disdat/apply.py]: No longer a string, but the dictionary itself.

File: disdat/apply.py [hunk 2]
```diff
@@ -51,7 +51,7 @@ def apply(output_bundle, pipe_params, pipe_cls, input_tags, output_tags, force,
...
-        pipe_cls:      String <module.ClassName>
+        pipe_cls (type[PipeTask]):      reference to the task class
```
Comment (kyocum) [inline disdat/apply.py]: So I know we can now declare types in P3.   But I haven't seen this syntax for types in google doc strings.   I usually say `disdat.pipe.PipeTask` for the type.

File: disdat/api.py [hunk 3]
```diff
@@ -1016,7 +1018,7 @@ def apply(local_context, transform, output_bundle='-',
 
     Args:
         local_context (str):  The name of the local context in which the pipeline will run in the container
-        transform (str): The name of the Disdat Pipe class, typically `<package>.<module>.<class>`
+        transform (type[PipeTask]): A reference to the Disdat Pipe class
```
Comment (kyocum) [inline disdat/api.py]: (`disdat.pipe.PipeTask`) instead?

File: disdat/common.py [hunk 4]
```diff
@@ -167,7 +168,7 @@ def _read_configuration_file(self, disdat_config_file, luigi_config_file):
         Next, see if there is a disdat.cfg in cwd.  Then configure disdat and (re)configure logging.
         """
         # _logger.debug("Loading config file [{}]".format(disdat_config_file))
-        config = configparser.SafeConfigParser({'meta_dir_root': self.meta_dir_root, 'ignore_code_version': 'False'})
+        config = configparser.ConfigParser({'meta_dir_root': self.meta_dir_root, 'ignore_code_version': 'False'})
```
Comment (kyocum) [inline disdat/common.py]: because p3, OK.

File: disdat/apply.py (hunk 2 above)
Comment (jcessna) [reply disdat/apply.py]: i can change it to that, but technically the (type) in the parentheses is the thing returned by type(value).  by putting disdat.pipe.PipeTask, you are implying that the value should be an INSTANCE of that class.  

this type[class] notation is understood by pycharm, so i believe it is standard.

perhaps it should be type[disdat.pipe.PipeTask] to be more explicit.

thoughts?

File: disdat/common.py (hunk 4 above)
Comment (jcessna) [reply disdat/common.py]: yeah, SaveConfigParser is deprecated

File: disdat/driver.py [hunk 5]
```diff
@@ -198,19 +146,19 @@ def requires(self):
...
-            task_params.update(json.loads(self.pipe_params))
-
-        t = DriverTask.inflate_cls(self.pipe_cls, task_params)
+        if self.pipe_params:
+            task_params.update(self.pipe_params)
 
-        return t
+        # Instantiate and return the class directly with the parameters
+        # Instance caching is taken care of automatically by luigi
+        return self.pipe_cls(**task_params)
```
Comment (kyocum) [inline disdat/driver.py]: Let's change the comment above for the type of pipe_cls to class object.

File: disdat/dsdt.py [hunk 6]
```diff
@@ -104,7 +104,7 @@ def main():
     apply_p.add_argument('-f', '--force', action='store_true', help="If there are dependencies, force re-computation.")
     apply_p.add_argument('--incremental-push', action='store_true', help="Commit and push each task's bundle as it is produced to the remote.")
     apply_p.add_argument('--incremental-pull', action='store_true', help="Localize bundles as they are needed by downstream tasks from the remote.")
-    apply_p.add_argument('pipe_cls', type=str, help="User-defined transform, e.g., 'module.PipeClass'")
+    apply_p.add_argument('pipe_cls', type=load_class, help="User-defined transform, e.g., 'module.PipeClass'")
```
Comment (kyocum) [inline disdat/dsdt.py]: Is it pretty to have a type be a method?  It's opaque.  Can we leave at as str and do this in apply.main() and maybe rename main?  Note that, b/c we expressly called task_register code previously, we no longer get a list of possibly task candidates as we once did when someone enters in a bogus value.    
`dsdt apply swordfish.monger` used to give:
```
luigi.task_register.TaskClassNotFoundException: No task swordfish. Candidates are: BundleWrapperTask,Config,DriverTask,ExternalTask,PipeTask,RangeBase,RangeByMinutes,RangeByMinutesBase,RangeDaily,RangeDailyBase,RangeHourly,RangeHourlyBase,RangeMonthly,Task,TestNotificationsTask,WrapperTask,batch_email,core,email,execution_summary,scheduler,sendgrid,smtp,worker
```
but now gives
`ModuleNotFoundError: No module named 'swordfishasd'`

File: disdat/hyperframe.py [hunk 7]
```diff
@@ -48,6 +48,7 @@
 import tempfile
 from datetime import datetime
 import uuid
+import json
```
Comment (kyocum) [inline disdat/hyperframe.py]: we can now remove this import

File: disdat/apply.py (hunk 2 above)
Comment (kyocum) [reply disdat/apply.py]: Got it.   The small change then to the fully qualified type[disdat.pipe.PipeTask] sounds perfect.

File: disdat/api.py (hunk 3 above)
Comment (kyocum) [reply disdat/api.py]: `type[disdat.pipe.PipeTask]`

**Comment by kyocum** [comment]
Small comments / changes, many to doc strings.   Otherwise, nice PR!

**Pull request merged**